### PR TITLE
Add: Unit testing

### PR DESCRIPTION
**Pull request by Bavon101**
As per this PR my partner and I achieved the following:

- [x] We had to set up  React Testing Library.
- [x] Write unit tests for the files operate.js and calculate.js using Jest.
- [x] Created unit tests for all React components.
- [x] Use React Testing Library to simulate user interaction (Calculator actions)
- [x] All tests are passing.

### CODE REVIEW

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @Bavon101  and @mnekx , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, where linters are not set up. Kindly add them and ask for another code review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/App.test.js [hunk 1]
```diff
@@ -1,8 +1,84 @@
+import renderer from 'react-test-renderer';
```
Comment (GzimAsani) [inline src/App.test.js:1]: - [x] One of your tests is failing, kindly fix it:
Screenshot from 2022-03-23 22-38-36

File: src/App.test.js [hunk 2]
```diff
@@ -1,8 +1,84 @@
...
+  const number2 = screen.getByText('5');
+  const operan = screen.getByText('+');
+  const equal = screen.getByText('=');
+  userEvent.click(number1);
+  userEvent.click(operan);
+  userEvent.click(number2);
+  userEvent.click(equal);
+  expect(container.getElementsByClassName('math-box')[0].textContent).toBe(
+    '11',
+  );
```
Comment (GzimAsani) [inline src/App.test.js]: - [x] All the functionalities & scenarios of the operate.js and calculate.js modules should be covered using Jest. _Currently you are testing operations only with +, please do the same with others_

Comment (GzimAsani) [reply src/App.test.js]: - [x] There should be the usage of Jest snapshots for testing all the components.

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi  @Bavon101 and @mnekx      
Good job so far! You are almost done :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check the comments below :arrow_double_down: _

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
Hi @Bavon101, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**